pkg/ccl/backupccl/testdata/backup-restore/backup-dropped-descriptors-declarative

```diff
@@ -110,31 +111,35 @@ SELECT orig->'type'->'name', orig->'type'->'state' FROM tbls WHERE id = 110 OR i
 "typ" "OFFLINE"
 "_typ" "OFFLINE"
 
-# A database backup should succeed and should include the offline schema,
-# type, and table.
+# A database backup should succeed and should include the offline schema and type but NOT the
+# offline table t.
 exec-sql
 BACKUP DATABASE d2 INTO 'nodelocal://0/dropped-schema-in-database';
 ----
 
 query-sql
-SELECT count(*)
+SELECT object_name
   FROM [SHOW BACKUP LATEST IN 'nodelocal://0/dropped-schema-in-database']
   WHERE object_name IN ('s', 't', 'typ', '_typ');
 ----
-4
+typ
+_typ
+s
 
 # A cluster backup should succeed but should include the offline schema,
-# type, and table.
+# type, but not the offline table.
 exec-sql
 BACKUP INTO 'nodelocal://0/cluster/dropped-schema-in-database';
 ----
 
 query-sql
-SELECT count(*)
+SELECT object_name
   FROM [SHOW BACKUP LATEST IN 'nodelocal://0/cluster/dropped-schema-in-database']
   WHERE object_name IN ('s', 't', 'typ', '_typ');
 ----
-4
+typ
+_typ
+s
 
```

Comment:
Perhaps this PR doesn't go far enough: as you see above,`typ` and `_typ` are offline type descriptors, and `s` is an offline schema and both are included in the (cluster) backup, because as of https://github.com/cockroachdb/cockroach/pull/83915, the drop job moves these guys briefly into an offline state. 

That _seems_ fine, as long as the online schema changer follows through and drops these descriptors in RESTORE. In other words, I _think_ the restoring cluster should not see schema s. But below, the test _expects_ to see offline schema s in the restored database. That seems wrong? Or maybe the backed up offline schema is live in the restored cluster because the restore job rolls back the drop? But is that also incorrect???

Reply:
Types and schemas can't reference tables so it should be OK, however, tables can. I suggest you add a test case (if there isn't one already) with views which reference a type and a table that are also being dropped, and see whether the schema change chugs along post-restore.

Reply:
Why do you think this is OK?

`typ` (type desc)  and `s` (schema desc) were in the middle of getting dropped when the backup occurred. This test suggests that these descriptors are now _online_ in the restored cluster.  I think these descriptors should not exist in the restored cluster, as they were getting dropped in the when the backup was taken. But maybe i'm misunderstanding something?

Reply:
I thought you can't drop a table if a view depends on it?

Reply:
resolved offline: andrew will refactor the schema changer to not use the offline state. Down the road, restore/import will create a new descriptor state for IMPORTING and RESTORING.